paper/src/main/java/vg/civcraft/mc/civmodcore/world/locations/chunkmeta/WorldChunkMetaManager.java

```diff
@@ -56,8 +56,14 @@ public WorldChunkMetaManager(World world, short worldID) {
 			}
 			return a.compareTo(b);
 		}));
+
+		this.chunkLoadingQueue = new LinkedBlockingQueue<>();
```

Comment:
Not sure about this one.
I prefer the following algorithm:
1. If the class doesn't have an explicit constructor then I declare initialization at the place of fields
2. If the class has an explicit constructor I prefer to initialize fields in the constructor

Also, exactly in this case it won't raise null reference exception because I declared all fields as **final**.

Reply:
Number 2 will create null references if a new constructor is introduced. It's safer to construct _primitive_ fields in place, if they require no logic. Collection can be constructed on field declaration, because its safe and easy. No need for the constructor to handle it. If another constructor is introduced at some point, the fields may be nulled, because calling the other constructor is not required in java.

```java
...
final List<Integer> something;

public Constructor1() {
    something = new ArrayList();
}

public Constructor2(int arg1) {
   Constructor1(); // Correct.
   this.intfield = arg1;
}

public Constructor3(String arg1) {
   this.field1 = arg1;
   // Incorrect, something is now a null reference because default field initialization.
}
```